frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts

```diff
@@ -97,11 +96,11 @@ export class ImportExperimentComponent implements OnInit {
     // Set loading to true before processing the files
     this.isLoadingExperiments$ = true;
 
-    const readFile = (fileIndex) => {
+    const readFile = async (fileIndex) => {
```

Comment:
This is loading the files one after another. You can use `map` on `event.target.files` over here to load them together.

Reply:
code updated